test/e2e/predictor/test_autoscaling.py

```diff
@@ -0,0 +1,151 @@
+    kserve_client.wait_isvc_ready(service_name, namespace=KSERVE_TEST_NAMESPACE)
+    pods = kserve_client.core_api.list_namespaced_pod(KSERVE_TEST_NAMESPACE,
+                                                      label_selector='serving.kserve.io/inferenceservice={}'
+                                                      .format(service_name))
+
+    annotations = pods.items[0].metadata.annotations
+
+    assert(annotations['autoscaling.knative.dev/metric'] == 'concurrency')
+    assert(annotations['autoscaling.knative.dev/target'] == '1')
+    res = predict(service_name, "./data/iris_input.json")
+    assert res["predictions"] == [1, 1]
+
+    kserve_client.delete(service_name, KSERVE_TEST_NAMESPACE)
+
+
+def test_sklearn_kserve_rps():
+    service_name = "isvc-sklearn-scale-rps"
+    predictor = V1beta1PredictorSpec(
+        min_replicas=1,
+        scale_metric='rps',
+        scale_target=5,
+        sklearn=V1beta1SKLearnSpec(
+            storage_uri="gs://kfserving-examples/models/sklearn/1.0/model",
+            resources=V1ResourceRequirements(
+                requests={"cpu": "100m", "memory": "256Mi"},
+                limits={"cpu": "100m", "memory": "256Mi"},
+            ),
+        ),
+    )
+
+    isvc = V1beta1InferenceService(
+        api_version=constants.KSERVE_V1BETA1,
+        kind=constants.KSERVE_KIND,
+        metadata=client.V1ObjectMeta(
+            name=service_name, namespace=KSERVE_TEST_NAMESPACE
+        ),
+        spec=V1beta1InferenceServiceSpec(predictor=predictor),
+    )
+
+    kserve_client = KServeClient(config_file=os.environ.get("KUBECONFIG", "~/.kube/config"))
+    kserve_client.create(isvc)
+    kserve_client.wait_isvc_ready(service_name, namespace=KSERVE_TEST_NAMESPACE)
+    pods = kserve_client.core_api.list_namespaced_pod(KSERVE_TEST_NAMESPACE,
+                                                      label_selector='serving.kserve.io/inferenceservice={}'
+                                                      .format(service_name))
+
+    annotations = pods.items[0].metadata.annotations
+
+    assert(annotations['autoscaling.knative.dev/metric'] == 'rps')
+    assert(annotations['autoscaling.knative.dev/target'] == '5')
+    res = predict(service_name, "./data/iris_input.json")
+    assert res["predictions"] == [1, 1]
+    kserve_client.delete(service_name, KSERVE_TEST_NAMESPACE)
+
+
+def test_sklearn_kserve_cpu():
+    service_name = "isvc-sklearn-scale-cpu"
+    predictor = V1beta1PredictorSpec(
+        min_replicas=1,
+        scale_metric='cpu',
+        scale_target=50,
+        sklearn=V1beta1SKLearnSpec(
+            storage_uri="gs://kfserving-examples/models/sklearn/1.0/model",
+            resources=V1ResourceRequirements(
+                requests={"cpu": "100m", "memory": "256Mi"},
+                limits={"cpu": "100m", "memory": "256Mi"},
+            ),
+        ),
+    )
+
+    annotations = dict()
+    annotations['autoscaling.knative.dev/class'] = 'hpa.autoscaling.knative.dev'
```

Comment:
Can we add another test case for raw deployment? for raw deployment mode we should use HPA.